src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -101,7 +102,6 @@ public GoogleLoginState(
 
     this.isLoggedIn = false;
     this.email = "";
-    connected = true; // assume we're connected until checkCredentials is called
 
     listeners = Lists.newLinkedList();
```

Comment:
I figured it wasn't thread-safe looking at the rest.  The listeners are a special case though and might be worth addressing here.  I think a ConcurrentLinkedDeque would solve it quickly.